src/ComponentDiscovery.php

```diff
@@ -98,10 +89,17 @@ public function getAll() {
    *   Array of extension objects for top-level Lightning components.
    */
   public function getMainComponents() {
-    $base_path = $this->getBaseComponentPath();
-
-    $filter = function (Extension $module) use ($base_path) {
-      return dirname($module->getPath()) == $base_path;
+    $identifier = 'lightning_';
+
+    $filter = function (Extension $module) use ($identifier) {
+      // Assumes that:
+      // 1. Lightning sub-components are always in a sub-directory within the
+      //    main component.
+      // 2. The main component's directory starts with "lightning_".
+      // E.g.: "/lightning_core/modules/lightning_search".
+      $path = explode('/', $module->getPath());
+      $parent = $path[count($path)-3];
```

Comment:
What if $path has less than 3 elements already?

Reply:
Good question. If that's the case, the extension must be directly in the "/modules" directory which would mean it's definitely not a sub-component. In that case, $parent would end up getting set to `null` which is in turn used as the haystack in `strpos()` - which is used to determine if this is a sub-component - which is definitely is not! :)

Obviously not very self-documenting :/ agree this should probably be improved.